### PR TITLE
arrow: add version 12.0.0

### PR DESCRIPTION
**Pull request by raulcd**
Specify library name and version:  **arrow/12.0.0**

On the Arrow 12.0.0 release [PLASMA was removed](https://github.com/apache/arrow/issues/33243) and [ACERO was moved to its own shared library](https://github.com/apache/arrow/issues/15280).

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/arrow/all/conanfile.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ class ArrowConan(ConanFile):
         "parquet": ["auto", True, False],
         "substrait": [True, False],
         "skyhook": [True, False],
-        "plasma": [True, False],
```
Comment (RubenRBS) [inline recipes/arrow/all/conanfile.py:29]: See https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#can-i-remove-an-option-from-a-recipe for the guideline on how to handle cases like this :)

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot!
I have one comment regarding the option deletion, but the rest looks good!

File: recipes/arrow/all/conanfile.py (hunk 1 above)
Comment (raulcd) [reply recipes/arrow/all/conanfile.py:29]: Thanks @RubenRBS !

**Comment by raulcd** [comment]
@RubenRBS do you know if there's anything else to be done here? I can't see the CI checks that errored, I get a 403 Forbidden on these links: 
* https://ci-conan-prod.jfrog.team/job/prod/job/cci/job/PR-17540/2/display/redirect
* https://ci-conan-prod.jfrog.team/job/prod-v2/job/cci/job/PR-17540/2/display/redirect

**Comment by RubenRBS** [comment]
The error is before the build so the bot is not posting the logs :)

The error is
```
ERROR: arrow/8.0.0: Error in package_id() method, line 545
	self.info.options.with_gflags = self._with_gflags()
while calling '_with_gflags', line 260
	if required or self.options.with_gflags == "auto":
	ConanException: 'self.options' access in 'package_id()' method is forbidden
```

**Comment by raulcd** [comment]
@RubenRBS this seems to have been happening in the past too, see:
* https://github.com/conan-io/conan-center-index/pull/15420
* https://github.com/conan-io/conan-center-index/pull/15696
* https://github.com/conan-io/conan-center-index/pull/17458

Should we fix that on a different PR?

**Review by mpimenov — DISMISSED** [review]
I think the conan 2 build error should be fixed in a different PR and this one is good as is.

File: recipes/arrow/all/conanfile.py [hunk 2]
```diff
@@ -579,12 +583,15 @@ def package_info(self):
             self.cpp_info.components["libarrow_substrait"].names["pkg_config"] = "arrow_substrait"
             self.cpp_info.components["libarrow_substrait"].requires = ["libparquet", "dataset"]
 
-        if self.options.plasma:
-            self.cpp_info.components["libplasma"].libs = [self._lib_name("plasma")]
-            self.cpp_info.components["libplasma"].names["cmake_find_package"] = "plasma"
-            self.cpp_info.components["libplasma"].names["cmake_find_package_multi"] = "plasma"
-            self.cpp_info.components["libplasma"].names["pkg_config"] = "plasma"
-            self.cpp_info.components["libplasma"].requires = ["libarrow"]
+        # Plasma was deprecated on Arrow 12.0.0
```
[the remark this answers is not in the document]
Comment (raulcd) [reply recipes/arrow/all/conanfile.py]: Thanks, done!

**Review by RubenRBS — APPROVED** [review]
Let's fix the issue in another PR as @mpimenov mentioned! :) Thanks a lot for your patience

**Review by memsharded — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**